sdks/java/core/src/main/java/org/apache/beam/sdk/io/Read.java

```diff
@@ -743,8 +727,6 @@ public boolean tryClaim(UnboundedSourceValue<OutputT>[] position) {
               currentReader.close();
             } catch (IOException closeException) {
               e.addSuppressed(closeException);
-            } finally {
```

Comment:
Intended?

Reply:
Yes, we keep the reader around and let the gc mechanism get rid of it.